.devcontainer/Dockerfile

```diff
@@ -0,0 +1,50 @@
+FROM ubuntu
+
+RUN apt-get update && apt-get upgrade
+
+RUN apt-get install --yes \
+        bash-completion \
+        build-essential \
+        curl \
+	cmake \
+        direnv \
+        emacs-nox \
+        htop \
+        jq \
+        less \
+        lsof \
+        man-db \
+        nano \
+        neovim \
+        ssl-cert \
+        sudo \
+        unzip \
+        zip
+
+RUN apt-get install locales && locale-gen en_US.UTF-8
+ENV LANG=en_US.UTF-8
+
+RUN bash -c "$(curl https://nixos.org/nix/install)"
+
+RUN bash -c "$(curl -fsSL https://deb.nodesource.com/setup_14.x)" \
+	&& apt-get install -y nodejs
+
+RUN bash -c "$(curl -fsSL https://raw.githubusercontent.com/horta/zstd.install/main/install)"
+
+RUN echo 'deb [trusted=yes] https://repo.goreleaser.com/apt/ /' | sudo tee /etc/apt/sources.list.d/goreleaser.list \
+	&& apt update \
+	&& apt install nfpm
+
+
+ENV GO_VERSION=1.19
+ENV GOPATH=$HOME/go-packages
+ENV GOROOT=$HOME/go
+ENV PATH=$GOROOT/bin:$GOPATH/bin:$PATH
+RUN curl -fsSL https://dl.google.com/go/go$GO_VERSION.linux-amd64.tar.gz | tar xzs
+
+RUN echo 'export PATH=$GOPATH/bin:$PATH' >> $HOME/.bashrc
```

Comment:
After installing nix, we _shouldn't_ need to add anything else I believe, although I know this is in draft ;p

Reply:
Agreed; I don't want to _dictate_ people use nix however much I want folks to _just use nix_.